pkg/pagerduty/service.go

```diff
@@ -365,39 +367,35 @@ func (c *SvcClient) UpdateEscalationPolicy(data *Data) error {
 	return nil
 }
 
+// resolvePendingIncidents loops over all unresolved incidents to resolve all contained alerts
 func (c *SvcClient) resolvePendingIncidents(data *Data) error {
-
 	incidents, err := c.getIncidents(data)
 	if err != nil {
 		return err
 	}
 
-	if len(incidents) > 0 {
-		serviceKey, err := c.GetIntegrationKey(data)
+	for _, incident := range incidents {
+		alerts, err := c.PdClient.ListIncidentAlerts(incident.Id)
 		if err != nil {
 			return err
 		}
 
-		for _, incident := range incidents {
-			alerts, err := c.PdClient.ListIncidentAlerts(incident.Id)
+		for _, alert := range alerts.Alerts {
+			err = c.resolveAlert(alert.Integration.ID, alert.AlertKey)
```

Comment:
Are you sure `alert.Integration.ID` is the right value to pass in to the events API as routing key?
From reading the [Events API reference](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODI2Nw-send-an-event-to-pager-duty), it looks like the routing key is the integration key:
> The GUID of one of your Events API V2 integrations. This is the "Integration Key" listed on the Events API V2 integration's detail page.
Example value (see example on the right side):
```
  "routing_key": "e93facc04764012d7bfb002500d5d1a6",
```

while the Integration.ID seems to be the ID of an integration, comparing above example value to the example from the alerts [API reference](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODE0Mw-list-alerts-for-an-incident):
```
      "integration": {
        "id": "PQ12345",
```
Would we probably need to transform it into a integration key like done [here](https://github.com/openshift/pagerduty-operator/blob/01dfafecc82a7713dac35f103caa2702a90eb687/pkg/pagerduty/service.go#L212)?

Reply:
Yes definitely, good catch! I even renamed the variable from `serviceKey` to `integrationKey`, but still made this mistake haha